tests/kda/test_kda_output_only.py

```
@@ -0,0 +1,386 @@
+    if not _KDA_OUTPUT_ONLY_AVAILABLE:
+        pytest.skip("kda_output_only_decode unavailable (missing cutlass DSL deps)")
+
+
+def naive_output_only_reference(q, k, v, g, beta, h0_pool, idx, scale):
+    """fp32 recurrent reference matching the kernel contract.
+
+    q/k [B,T,H,K] (caller pre-applies L2 norm), v [B,T,HV,V],
+    g [B,T,HV,K] log-space fp32, beta [B,T,HV] fp32 (post-sigmoid),
+    h0_pool [pool,HV,V,K] bf16, idx [B] int32. Returns o [B,T,HV,V] fp32.
+    """
+    B, T, H, K = q.shape
```

Ruff (0.16.2)

[warning] Unpacked variable `K` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)